packages/langium/src/references/name-provider.ts

```diff
@@ -21,9 +21,16 @@ export function isNamed(node: AstNode): node is NamedAstNode {
 export interface NameProvider {
     /**
      * Returns the `name` of a given AstNode.
-     * @param node Specified `AstNode` whose name node shall be retrieved.
+     * @param node Specified `AstNode` whose name shall be retrieved.
      */
     getName(node: AstNode): string | undefined;
+
+    /**
+     * Returns the property name that is used to store the name of an AstNode.
+     * @param node The AstNode for which the name property shall be retrieved.
+     */
+    getNameProperty(node: AstNode): string | undefined;
```

Comment:
asking myself: why is the property name for storing the ast node name dynamic?

Reply:
Because some developers prefer not using `name` as the main property for giving an element a name. For example, some devs use `id` or something similar. The `NameProvider` in general is just there as a way for adopters to override how you retrieve the name property from an AST node.